src/pages/proposal.tsx

```diff
@@ -57,59 +65,112 @@ import './proposals.css'
+const EMPTY_CHOICE_SELECTION: SelectedChoice = { choice: undefined, choiceIndex: undefined }
 const PROPOSAL_STATUS_WITH_UPDATES = new Set([ProposalStatus.Passed, ProposalStatus.Enacted])
+const EMPTY_CHOICES: string[] = []
+const MAX_ERRORS_BEFORE_SNAPSHOT_REDIRECT = 3
+const SECONDS_FOR_VOTING_RETRY = 5
 
-type ProposalPageOptions = {
-  changing: boolean
+export type ProposalPageContext = {
+  changingVote: boolean
   confirmSubscription: boolean
   confirmDeletion: boolean
   confirmStatusUpdate: ProposalStatus | false
   showVotesList: boolean
   showProposalSuccessModal: boolean
   showUpdateSuccessModal: boolean
+  showVotingModal: boolean
+  showVotingError: boolean
+  showSnapshotRedirect: boolean
+  retryTimer: number
+  selectedChoice: SelectedChoice
 }
 
 export default function ProposalPage() {
   const t = useFormatMessage()
   const location = useLocation()
   const params = useMemo(() => new URLSearchParams(location.search), [location.search])
-  const [options, patchOptions] = usePatchState<ProposalPageOptions>({
-    changing: false,
+  const [proposalContext, updateContext] = usePatchState<ProposalPageContext>({
+    changingVote: false,
     confirmSubscription: false,
     confirmDeletion: false,
     confirmStatusUpdate: false,
     showVotesList: false,
     showProposalSuccessModal: false,
     showUpdateSuccessModal: false,
+    showVotingModal: false,
+    showVotingError: false,
+    showSnapshotRedirect: false,
+    retryTimer: SECONDS_FOR_VOTING_RETRY,
+    selectedChoice: EMPTY_CHOICE_SELECTION,
   })
-  const patchOptionsRef = useRef(patchOptions)
+  const [errorCounter, setErrorCounter] = useState(0)
+  const updateContextRef = useRef(updateContext)
   const [account, { provider }] = useAuthContext()
   const [proposal, proposalState] = useProposal(params.get('id'))
   const { isCommittee } = useIsCommittee(account)
   const { votes, votesState } = useProposalVotes(proposal?.id)
+  const { surveyTopics, isLoadingSurveyTopics } = useSurveyTopics(proposal?.id)
   const [subscriptions, subscriptionsState] = useAsyncMemo(
     () => Governance.get().getSubscriptions(proposal!.id),
     [proposal],
     { callWithTruthyDeps: true }
   )
+  const choices: string[] = proposal?.snapshot_proposal?.choices || EMPTY_CHOICES
+  const partialResults = useMemo(() => calculateResult(choices, votes || {}), [choices, votes])
 
   const { publicUpdates, pendingUpdates, nextUpdate, currentUpdate, refetchUpdates } = useProposalUpdates(proposal?.id)
 
   const subscribed = useMemo(
     () => !!account && !!subscriptions && !!subscriptions.find((sub) => sub.user === account),
     [account, subscriptions]
   )
-  const [voting, vote] = useAsyncTask(
-    async (_: string, choiceIndex: number) => {
-      if (proposal && account && provider && votes) {
+
+  const proposalResults = useRef<HTMLDivElement>(null)
+
+  const handleScrollClick = () => {
+    proposalResults.current?.scrollIntoView({ behavior: 'smooth' })
```

Comment:
It looks like options param in `scrollIntoView` are not supported on Safari. https://developer.mozilla.org/es/docs/Web/API/Element/scrollIntoView
Is that right? Are we ok with that?

Reply:
Ty, just tested this on Safari and it works :)